cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetService.java

```diff
@@ -73,12 +78,29 @@ public DatasetService(CConfiguration cConf,
                         DatasetOpExecutor opExecutorClient,
                         Set<DatasetMetricsReporter> metricReporters,
                         DatasetTypeService datasetTypeService,
-                        DatasetInstanceService datasetInstanceService) throws Exception {
+                        DatasetInstanceService datasetInstanceService) {
     this.cConf = cConf;
     this.typeService = datasetTypeService;
     DatasetTypeHandler datasetTypeHandler = new DatasetTypeHandler(datasetTypeService);
     DatasetInstanceHandler datasetInstanceHandler = new DatasetInstanceHandler(datasetInstanceService);
     NettyHttpService.Builder builder = new CommonNettyHttpServiceBuilder(cConf, Constants.Service.DATASET_MANAGER)
+      .addChannelPipelineModifier(new ChannelPipelineModifier() {
+        @Override
+        public void modify(ChannelPipeline channelPipeline) {
+          if (LOG.isTraceEnabled()) {
```

Comment:
Might be better not to add the modifier if log trace is not enabled.

Reply:
I was thinking that with Twill, we change the log level after a container has started. Do you think it is a performance concern?

Reply:
That's all right, shouldn't be too big of a concern.